### PR TITLE
various fix after first tests

### PR DESCRIPTION
**Pull request by dsavina**
no new features, only corrections (variable names, wrong parameters, uncatched exceptions, etc.)

### CODE REVIEW

File: Api/v1/Models/AbstractTemplate.php [hunk 1]
```diff
@@ -82,6 +83,8 @@ public function __construct(FileService $fileService, string $contentType, int $
     public function download()
     {
         $this->template = $this->fileService->downloadFile($this->fileService->generateRandomFileName($this->templateFileExtension), $this->templateUrl);
+        if ($this->template === null)
+            throw new FileNotFoundException('impossible de charger les templates distants fournis dans la requête', 404);
```
Comment (gulien) [inline Api/v1/Models/AbstractTemplate.php]: Please use the following format: "File not found exception: unable to download remote templates."

File: Api/v1/Services/MimeTypes.php [hunk 2]
```diff
@@ -0,0 +1,324 @@
+<?php
+namespace Api\v1\Services;
+
+class MimeType
```
Comment (gulien) [inline Api/v1/Services/MimeTypes.php]: Please rename this class "MimeTypeService" and comment it and its methods!

File: Api/v1/Services/MimeTypes.php [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+{
+    public static function getExtensionFromMimetype($mimeType)
+    {
+        if (array_key_exists($mimeType, MIMETIC_TO_EXTENSION))
```
Comment (gulien) [inline Api/v1/Services/MimeTypes.php]: Please use brackets.

File: Api/v1/Models/AbstractTemplateToPopulate.php [hunk 4]
```diff
@@ -59,6 +59,7 @@ public function __construct(FileService $fileService, string $contentType, int $
     function __destruct()
```
Comment (gulien) [inline Api/v1/Models/AbstractTemplateToPopulate.php:59]: Hmm maybe AbstractTemplate should implements __destruct() and then AbstractTemplateToPopulate should call it. In that way, AbstractTemplate will handle the deletion of the template and AbstractTemplateToPopulate the deletion of the... popuplatedTemplate.

**Comment by dsavina** [comment]
All fixed for merge !

**Pull request merged**